File: backend/src/main/java/org/example/backend/model/Event.java
package org.example.backend.model;

import lombok.With;

import java.time.LocalDateTime;

@With
public record Event(
        String id,
        String title,
        EventCategory category,
        String description,
        String address,
        String city,
        String country,
        LocalDateTime date
) {
}

File: frontend/src/model/Event.ts
export type Event = {
    title: string,
    category: "RESTAURANT" | "COFFEE" | "BAR" | "BAKERY" | "THINGS_TO_DO" | "EVENT" | "HOTEL" | "TRANSPORT" | "MEETING" | "NOTE",
    description: string,
    address: string,
    city: string,
    country: string,
    date: string
}